core/include/tangram/tile/tileID.h

```diff
@@ -45,28 +45,33 @@ struct TileID {
         return isValid() && z <= _maxZoom;
     }
 
-    TileID withMaxSourceZoom(int32_t _max) const {
+    TileID withMaxSourceZoom(int32_t _maxZoom) const {
 
-        if (z <= _max) {
+        if (z <= _maxZoom) {
             return *this;
         }
 
-        int32_t over = z - _max;
+        int32_t over = z - _maxZoom;
 
-        return TileID(x >> over, y >> over, _max, z, wrap);
+        return TileID(x >> over, y >> over, _maxZoom, s, wrap);
     }
 
-    TileID getParent() const {
+    TileID scaled(int32_t _tileScale) const {
+        if (_tileScale <= 0) { return *this; }
```

Comment:
Actually, maybe it makes more sense for these functions that require parameters from the tile source to be members of `TileSource` itself, like `TileID TileSource::getParent(TileID tile)`. Not sure that it helps substantially for this PR, but something to keep in mind perhaps.